model/task/db.go

```diff
@@ -345,6 +345,22 @@ func ByVersion(version string) db.Q {
 	})
 }
 
+// AbortedTasksByVersion produces a query that returns all aborted tasks for the given version.
+func AbortedTasksByVersion(version string) db.Q {
+	return db.Query(bson.M{
+		VersionKey: version,
+		AbortedKey: true,
+	})
+}
+
+// FailedTasksByVersion produces a query that returns all failed tasks for the given version.
+func FailedTasksByVersion(version string) db.Q {
+	return db.Query(bson.M{
+		VersionKey: version,
+		StatusKey:  evergreen.TaskFailed,
```

Comment:
Should use $in evergreen.FailureTaskStatuses (and we should write a small unit test for at least one of these, with Count).